engine/utils/cli_selection_utils.h

```diff
@@ -20,7 +20,7 @@ inline std::optional<std::string> PrintSelection(
   std::string selection{""};
   PrintMenu(options);
   std::cout << "Select an option (" << 1 << "-" << options.size() << "): ";
-  std::cin >> selection;
+  std::getline(std::cin, selection);
```

Comment:
use std::getline to fix issue with interactive shell on `chat`
https://stackoverflow.com/questions/6819082/why-does-a-stdgetline-call-on-stdcin-not-wait-for-user-input